README.md
![Frete Rápido - Sistema Inteligente de Gestão Logística](https://freterapido.com/imgs/frete_rapido.png)

<hr>

### **Módulo para plataforma Magento**

Versão do módulo: 1.3.6

Compatibilidade com Magento: **1.9.x**

Links úteis:

- [Magento Marketplace][1]
- [Painel administrativo][2]
- [suporte@freterapido.com][3]


-------------

### IMPORTANTE

Este módulo é apenas um referencial de integração e cabe ao cliente a função de configurá-lo e adaptá-lo a sua respectiva loja, levando em conta as particularidades e conflitos que podem surgir durante o processo de integração.

A Frete Rápido não mantem e/ou oferece suporte para a integração com o **Magento**, disponibilizamos o módulo padrão que atente a modalidade de envio simples.

**Este módulo não opera Dropshipphig!**

Caso seja necessário adaptações deste módulo para atender a sua loja, é possível alterar o código fonte, desde que atenda a [API da Frete Rápido][9]. E [neste link][8] você encontra a documentação de orientações do **Magento**.

A Frete Rápido não se responsabiliza por eventualidades advindas deste módulo.

--------------

### Instalação

>**<i class="icon-attention"></i> ATENÇÃO!** Recomendamos que seja feito backup da sua loja antes de realizar qualquer instalação. A instalação desse módulo é de inteira responsabilidade do lojista.


- [Baixe a última versão aqui][4],  descompacte o conteúdo do arquivo zip dentro da pasta "app" da sua loja Magento.
- Verifique se o arquivo **local.xml** está habilitado na pasta "app/etc" da sua loja Magento. Caso não esteja, é só renomear o arquivo **local.xml.sample** para **local.xml**.
- Acesse a área administrativa de sua loja e limpe o cache em: Sistema > Gerenciamento de Cache.

![Mensagem de atenção para backup da loja](app/code/community/Freterapido/Freterapido/docs/img/attention_2.png "#FicaDica ;&#41;")

----------

# Documentação Frete Rápido

##  - [Configurações iniciais](https://freterapido.com/wiki/docs/integrations/magento_1.9/#configuracoes-iniciais).

1. [Configuração do módulo](https://freterapido.com/wiki/docs/integrations/magento_1.9/#_1-configuracao-do-modulo).

2. [Origem remetente](https://freterapido.com/wiki/docs/integrations/magento_1.9/#_2-origem-remetente).

3. [Medidas e Prazo de envio](https://freterapido.com/wiki/docs/integrations/magento_1.9/#_3-medidas-e-prazo-de-envio).

4. [Categorias](https://freterapido.com/wiki/docs/integrations/magento_1.9/#_4-categorias).

##  - [Contratação do Frete](https://freterapido.com/wiki/docs/integrations/magento_1.9/#contratacao-do-frete).

##  - [Cálculo do frete na página do produto](https://freterapido.com/wiki/docs/integrations/magento_1.9/#cotacoes-de-frete-na-pagina-do-produto).

##  - [Webhook](https://freterapido.com/wiki/docs/integrations/magento_1.9/#Webhook).

##  - [Considerações finais](https://freterapido.com/wiki/docs/integrations/magento_1.9/#consideracoes-finais).

### Contribuições
Encontrou algum bug ou tem sugestões de melhorias no código? Sensacional! Não se acanhe, nos envie um pull request com a sua alteração e ajude este projeto a ficar ainda melhor.

1. Faça um "Fork"
2. Crie seu branch para a funcionalidade: ` $ git checkout -b feature/nova-funcionalidade`
3. Faça o commit suas modificações: ` $ git commit -am "adiciona nova funcionalidade"`
4. Faça o push para a branch: ` $ git push origin feature/nova-funcionalidade`
5. Crie um novo Pull Request

--------

### Licença

[MIT][5]

[1]: https://marketplace.magento.com/freterapido-frete-rapido.html "Magento Connect"
[2]: https://painel.freterapido.com/?origin=doc_magento "Painel do Frete Rápido"
[3]: mailto:suporte@freterapido.com "Contato uma galera super gente fina :)"
[4]: https://github.com/freterapido/freterapido_magento/archive/master.zip
[5]: https://github.com/freterapido/freterapido_magento/blob/master/LICENSE
[6]: https://github.com/freterapido/freterapido_magento/blob/master/README.md
[7]: https://github.com/freterapido/freterapido_magento/blob/master/README_EN.md
[8]: http://devdocs.magento.com/
[9]: https://dev.freterapido.com


CHANGELOG.md
# Changelog

## [1.3.6] - 2019-09-13
### Adicionado

#### **Status customizáveis**

Quando configurados no módulo, ao contratar ou quando um frete for dado como entregue pela Frete Rápido, o módulo irá atualizar os respectivos pedidos para os status configurados.

#### **Canal de vendas para regra de frete**

A partir desta versão, é possível informar o canal de vendas na configuração do módulo, permitindo assim, a aplicação de regras de frete mais específicas.


#### **Categorias**

	- Acessório para decoração (com vidro)
	- Acessório para decoração (sem vidro)
	- Acessórios automotivos
	- Acessórios para bicicleta
	- Artesanatos (com vidro)
	- Bicicletas (desmontada)
	- Cama / Mesa / Banho
	- Chapas de madeira
	- Manequins
	- Portas / Janelas (com vidro)
	- Torneiras
	- Vasos de polietileno

#### **Webhook**

	- Inclusão dos status customizáveis para a atualização dos pedidos;

#### **Contratação do frete**

	- Inclusão dos status customizáveis para a atualização dos pedidos;

#### **Cotação do frete**

	- Envio do canal de venda;

### Alterado

#### **Categorias**

	- Artesanatos, -> Artesanatos (sem vidro),            
    - Portas / Janelas, -> Portas / Janelas (sem vidro),       
    - Materiais hidráulicos -> Materiais hidráulicos / Encanamentos

## [1.3.5] - 2019-08-29
### Adicionado

#### **Categorias**	

	- Utilidades domésticas;
	- Acessórios para celular;
	- Toldos;
	- Pisos cerâm / Revestimentos;
	- Artesanatos;
	- Quadros / Molduras;
	- Porta / Janelas;
	- Placa de Energia Solar;
	- Materiais hidráulicos;
	- Pia / Vasos;
	- Bijuteria;
	- Joia;
	- Refrigeração Industrial;
	- Cocção Industrial;
	- Utensílios industriais;
	- Maquina de algodão doce;
	- Maquina de chocolate;
	- Estufa térmica;
	- Equipamentos de cozinha industrial;
	- Tapeçaria / Cortinas / Persianas;

#### **Envio de NFe**
 	- A partir da versão 1.3.5 é possivel realizar o envio das NFes para que o rastreio possa ocorrer de forma efetiva pela Frete rápido.
		Para tal, se faz necessário realizar algumas ações:
		- Ao adicionar o faturamento no pedido, basta informar a data de emissão e chave de acesso em um comentário de faturamento seguindo o seguinte padrão: *nfe:17110752684209056803016340068050771046234656, emissao:29/08/2019 14:17:01*
		- Após adicionado o comentário seguindo a regra acima, tanto ao contratar quanto ao rastrear do frete, a nota fiscal informada será enviada automaticamente para a Frete Rápido 

### Alterado

#### **Webhook de ocorrências**
	- Implementação de retorno para testes do serviço
	- Correção do status HTTP de retorno


## [1.3.4] - 2019-06-27
### Adicionado

#### **Atributos**

Estes atributos servirão para indicar qual atributo será utilizado pelo módulo no momento da obtenção dos seus respectivos valores;

	- `ref_attr_height_type`-> Atributo referente à altura;
	- `ref_attr_width_type` -> Atributo referente à largura;
	- `ref_attr_length_type`-> Atributo referente ao comprimento;
	- `ref_attr_state_registration_type` -> Atributo referente à Insc. Estadual do destinatário;
	- Implementação de função para retorno dos valores dos produtos referenciados pelos atrbutos acima;
	- Implementação das classes necessárias para que os atributos sejam exibidos no front;

#### **Categorias**

	- Produto Químico Não Classificado;
	- Pilhas / Baterias;
	- Estiletes / Materiais cortantes;
	- Produto Químico Classificado;
	- Limpeza;
	- Extintores;
	- Equipamentos de segurança / EPI;

#### **Webhook**

	- Inclusão no arquivo de configuração do módulo, dos dados necessários para configurar a rota do webhook no qual a Frete Rápido irá utilizar para atualizar os status dos pedidos;
	- Inclusão do controller para manipular as requisições recebidas pela Frete Rápido;

### Alterado

#### **Categorias**

	- Produtos Químicos -> Produto Químico Não Classificado;

#### **Cotação / Contratação de frete**
	- Alterado o método de recuperação dos dados do destinatário do frete para obter, caso exista, dados como CNPJ e Inscrição Eestadual (este último quando informado o atributo customizável)
	- Implementação de filtros que permitam apenas números nos campos de CNPJ/CPF e Inscrição estadual do destinatário/remetente;
	- Alteração da imagem do MySql de mysql:5.6.23 para mariadb:10.1


## [1.3.3] - 2019-04-29
### Adicionado

#### **Categorias**

	- Alimentos não perecíveis;
    - Caixa de embalagem;
    - TV / Monitores;
    - Linha Branca;
    - Vitaminas / Suplementos nutricionais;
    - Malas / Mochilas;
    - Máquina / Equipamentos;
    - Rações / Alimento para Animal;
    - Artigos para Camping;

### Alterado

#### **Categorias**

	- Alimentos -> Alimentos perecíveis;
	- Cosméticos / Perfumaria-> Cosméticos;
	- Farmacêutico / Medicamentos -> Medicamentos;
	- Material Pet Shop / Rações -> Material Pet Shop;


app/code/community/Freterapido/Freterapido/Model/Source/Frcategory.php
<?php

/**
 * @category Freterapido
 * @package Freterapido_Freterapido
 * @author freterapido.com <suporte@freterapido.com>
 * @copyright Frete Rápido (https://freterapido.com)
 * @license https://github.com/freterapido/freterapido_magento/blob/master/LICENSE MIT
 */
class Freterapido_Freterapido_Model_Source_Frcategory extends Mage_Eav_Model_Entity_Attribute_Source_Abstract
{
    protected $_options = null;

    public function getAllOptions($withEmpty = false)
    {
        if (is_null($this->_options)) {
            $this->_options = [
                ['label' => 'Abrasivos',                            'value' => 1],
                ['label' => 'Adubos / Fertilizantes',               'value' => 2],
                ['label' => 'Alimentos perecíveis',                 'value' => 3],
                ['label' => 'Artigos para Pesca',                   'value' => 4],
                ['label' => 'Auto Peças',                           'value' => 5],
                ['label' => 'Bebidas / Destilados',                 'value' => 6],
                ['label' => 'Brindes',                              'value' => 7],
                ['label' => 'Brinquedos',                           'value' => 8],
                ['label' => 'Calçados',                             'value' => 9],
                ['label' => 'CD / DVD / Blu-Ray',                   'value' => 10],
                ['label' => 'Combustíveis / Óleos',                 'value' => 11],
                ['label' => 'Confecção',                            'value' => 12],
                ['label' => 'Cosméticos',                           'value' => 13],
                ['label' => 'Couro',                                'value' => 14],
                ['label' => 'Derivados Petróleo',                   'value' => 15],
                ['label' => 'Descartáveis',                         'value' => 16],
                ['label' => 'Editorial',                            'value' => 17],
                ['label' => 'Eletrônicos',                          'value' => 18],
                ['label' => 'Eletrodomésticos',                     'value' => 19],
                ['label' => 'Embalagens',                           'value' => 20],
                ['label' => 'Explosivos / Pirotécnicos',            'value' => 21],
                ['label' => 'Medicamentos',                         'value' => 22],
                ['label' => 'Ferragens',                            'value' => 23],
                ['label' => 'Ferramentas',                          'value' => 24],
                ['label' => 'Fibras Ópticas',                       'value' => 25],
                ['label' => 'Fonográfico',                          'value' => 26],
                ['label' => 'Fotográfico',                          'value' => 27],
                ['label' => 'Fraldas / Geriátricas',                'value' => 28],
                ['label' => 'Higiene',                              'value' => 29],
                ['label' => 'Impressos',                            'value' => 30],
                ['label' => 'Informática / Computadores',           'value' => 31],
                ['label' => 'Instrumento Musical',                  'value' => 32],
                ['label' => 'Liv',                                  'value' => 33],
                ['label' => 'Materiais Escolares',                  'value' => 34],
                ['label' => 'Materiais Esportivos',                 'value' => 35],
                ['label' => 'Materiais Frágeis',                    'value' => 36],
                ['label' => 'Material de Construção',               'value' => 37],
                ['label' => 'Material de Irrigação',                'value' => 38],
                ['label' => 'Material Elétrico / Lâmpa',            'value' => 39],
                ['label' => 'Material Gráfico',                     'value' => 40],
                ['label' => 'Material Hospitalar',                  'value' => 41],
                ['label' => 'Material Odontológico',                'value' => 42],
                ['label' => 'Material Pet Shop',                    'value' => 43],
                ['label' => 'Material Veterinário',                 'value' => 44],
                ['label' => 'Móveis montados',                      'value' => 45],
                ['label' => 'Moto Peças',                           'value' => 46],
                ['label' => 'Mudas / Plantas',                      'value' => 47],
                ['label' => 'Papelaria / Documentos',               'value' => 48],
                ['label' => 'Perfumaria',                           'value' => 49],
                ['label' => 'Material Plástico',                    'value' => 50],
                ['label' => 'Pneus e Borracharia',                  'value' => 51],
                ['label' => 'Produtos Cerâmicos',                   'value' => 52],
                ['label' => 'Produtos Químicos Não Classificados',  'value' => 53],
                ['label' => 'Produtos Veterinários',                'value' => 54],
                ['label' => 'Revistas',                             'value' => 55],
                ['label' => 'Sementes',                             'value' => 56],
                ['label' => 'Suprimentos Agrícolas / Rurais',       'value' => 57],
                ['label' => 'Têxtil',                               'value' => 58],
                ['label' => 'Vacinas',                              'value' => 59],
                ['label' => 'Vestuário',                            'value' => 60],
                ['label' => 'Vidros / Frágil',                      'value' => 61],
                ['label' => 'Cargas refrigeradas/congeladas',       'value' => 62],
                ['label' => 'Papelão',                              'value' => 63],
                ['label' => 'Móveis desmontados',                   'value' => 64],
                ['label' => 'Sofá',                                 'value' => 65],
                ['label' => 'Colchão',                              'value' => 66],
                ['label' => 'Travesseiro',                          'value' => 67],
                ['label' => 'Móveis com peças de vidro',            'value' => 68],
                ['label' => 'Acessórios de Airsoft / Paintball',    'value' => 69],
                ['label' => 'Acessórios de Pesca',                  'value' => 70],
                ['label' => 'Simulacro de Arma / Airsoft',          'value' => 71],
                ['label' => 'Arquearia',                            'value' => 72],
                ['label' => 'Acessórios de Arquearia',              'value' => 73],
                ['label' => 'Alimentos não perecíveis',             'value' => 74],
                ['label' => 'Caixa de embalagem',                   'value' => 75],
                ['label' => 'TV / Monitores',                       'value' => 76],
                ['label' => 'Linha Branca',                         'value' => 77],
                ['label' => 'Vitaminas / Suplementos nutricionais', 'value' => 78],
                ['label' => 'Malas / Mochilas',                     'value' => 79],
                ['label' => 'Máquina / Equipamentos',               'value' => 80],
                ['label' => 'Rações / Alimento para Animal',        'value' => 81],
                ['label' => 'Artigos para Camping',                 'value' => 82],
                ['label' => 'Pilhas / Baterias',                    'value' => 83],
                ['label' => 'Estiletes / Materiais Cortantes',      'value' => 84],
                ['label' => 'Produto Químico classificado',         'value' => 85],
                ['label' => 'Limpeza',                              'value' => 86],
                ['label' => 'Extintores',                           'value' => 87],
                ['label' => 'Equipamentos de Segurança / API',      'value' => 88],
                ['label' => 'Utilidades domésticas',                'value' => 89],
                ['label' => 'Acessórios para celular',              'value' => 90],
                ['label' => 'Toldos',                               'value' => 91],
                ['label' => 'Pisos cerâm / Revestimentos',          'value' => 92],
                ['label' => 'Artesanatos (sem vidro)',              'value' => 93],
                ['label' => 'Quadros / Molduras',                   'value' => 94],
                ['label' => 'Portas / Janelas (sem vidro)',         'value' => 95],
                ['label' => 'Placa de Energia Solar',               'value' => 96],
                ['label' => 'Materiais hidráulicos / Encanamentos', 'value' => 97],
                ['label' => 'Pia / Vasos',                          'value' => 98],
                ['label' => 'Bijuteria',                            'value' => 99],
                ['label' => 'Joia',                                 'value' => 100],
                ['label' => 'Refrigeração Industrial',              'value' => 101],
                ['label' => 'Cocção Industrial',                    'value' => 102],
                ['label' => 'Utensílios industriais',               'value' => 103],
                ['label' => 'Maquina de algodão doce',              'value' => 104],
                ['label' => 'Maquina de chocolate',                 'value' => 105],
                ['label' => 'Estufa térmica',                       'value' => 106],
                ['label' => 'Equipamentos de cozinha industrial',   'value' => 107],
                ['label' => 'Tapeçaria / Cortinas / Persianas',     'value' => 108],
                ['label' => 'Acessório para decoração (com vidro)', 'value' => 109],
                ['label' => 'Acessório para decoração (sem vidro)', 'value' => 110],
                ['label' => 'Acessórios automotivos',               'value' => 111],
                ['label' => 'Acessórios para bicicleta',            'value' => 112],
                ['label' => 'Artesanatos (com vidro)',              'value' => 113],
                ['label' => 'Bicicletas (desmontada)',              'value' => 114],
                ['label' => 'Cama / Mesa / Banho',                  'value' => 115],
                ['label' => 'Chapas de madeira',                    'value' => 116],
                ['label' => 'Manequins',                            'value' => 117],
                ['label' => 'Portas / Janelas (com vidro)',         'value' => 118],
                ['label' => 'Torneiras',                            'value' => 119],
                ['label' => 'Vasos de polietileno',                 'value' => 120],
                ['label' => 'Outros',                               'value' => 999],
            ];

            usort($this->_options, function ($a, $b) {
                return strcmp($a['label'], $b['label']);
            });
        }

        return $this->_options;
    }

    public function getOptionText($value)
    {
        $options = $this->getAllOptions(false);

        foreach ($options as $item) {
            if ($item['value'] == $value) {
                return $item['label'];
            }
        }

        return false;
    }
}
